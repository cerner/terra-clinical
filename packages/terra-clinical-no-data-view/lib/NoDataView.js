'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _NoDataIcon = require('./NoDataIcon');

var _NoDataIcon2 = _interopRequireDefault(_NoDataIcon);

require('./NoDataView.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  /**
   * The title of the no data view.
   */
  heading: _react.PropTypes.string,
  /**
   * The subtext description of the no data view.
   */
  subtext: _react.PropTypes.string,
  /**
   * The element to be dispaly in the subtext content container.
   */
  subtextContent: _react.PropTypes.element,
  /**
   * Whether or not the no data glyph should be displayed.
   */
  isGlyphHidden: _react.PropTypes.bool
};

var defaultProps = {
  heading: '',
  subtext: '',
  subtextContent: undefined,
  isGlyphHidden: false
};

var NoDataView = function NoDataView(_ref) {
  var heading = _ref.heading,
      subtext = _ref.subtext,
      subtextContent = _ref.subtextContent,
      isGlyphHidden = _ref.isGlyphHidden,
      customProps = _objectWithoutProperties(_ref, ['heading', 'subtext', 'subtextContent', 'isGlyphHidden']);

  var noDataClassNames = (0, _classnames2.default)(['terraClinical-NoDataView', customProps.className]);

  var glyphSection = void 0;
  if (!isGlyphHidden) {
    var noDataIcon = _react2.default.createElement(_NoDataIcon2.default, null);
    glyphSection = _react2.default.createElement(
      'div',
      { className: 'terraClinical-NoDataView-glyph' },
      noDataIcon
    );
  }

  var headingSection = void 0;
  if (heading) {
    headingSection = _react2.default.createElement(
      'p',
      { className: 'terraClinical-NoDataView-heading' },
      heading
    );
  }

  var subtextSection = void 0;
  if (subtext) {
    subtextSection = _react2.default.createElement(
      'p',
      { className: 'terraClinical-NoDataView-subtext' },
      subtext
    );
  }

  var subtextContentSection = void 0;
  if (subtextContent) {
    subtextContentSection = _react2.default.createElement(
      'div',
      { className: 'terraClinical-NoDataView-subtextContent' },
      subtextContent
    );
  }

  return _react2.default.createElement(
    'div',
    _extends({}, customProps, { className: noDataClassNames }),
    glyphSection,
    headingSection,
    subtextSection,
    subtextContentSection
  );
};

NoDataView.propTypes = propTypes;
NoDataView.defaultProps = defaultProps;

exports.default = NoDataView;