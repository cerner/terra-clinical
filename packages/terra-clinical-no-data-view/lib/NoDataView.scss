@import '~terra-mixins';

.terraClinical-NoDataView {
  align-items: center;
  display: flex;
  flex-direction: column;
  padding: 1.429rem 0;
  width: 100%;
}

.terraClinical-NoDataView-glyph {
  flex: 0 0 auto;
  padding-bottom: 0.2857rem;
  padding-top: 1.429rem;
}

.terraClinical-NoDataView-heading {
  @include terra-text-align-center();
  color: $terra-grey-25;
  flex: 0 0 auto;
  font-size: 1.7142857142857142rem;
  line-height: 1.284;
}

.terraClinical-NoDataView-subtext {
  @include terra-text-align-center();
  color: $terra-grey-65;
  flex: 0 0 auto;
  font-size: 1rem;
  line-height: 1.142;
  padding-top: 0.2857rem;
}

.terraClinical-NoDataView-subtextContent {
  flex: 0 0 auto;
  padding-top: 0.2857rem;
}

