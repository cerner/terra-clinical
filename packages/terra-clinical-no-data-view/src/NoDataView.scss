:local {
  .no-data-view {
    align-items: center;
    display: flex;
    flex-direction: column;
    padding: 1.429rem 0;
    width: 100%;
  }

  .glyph {
    flex: 0 0 auto;
    padding-bottom: 0.2857rem;
    padding-top: 1.429rem;
  }

  .heading {
    color: #c8cacb;
    flex: 0 0 auto;
    font-size: 1.7142857142857142rem;
    line-height: 1.284;
    margin-bottom: 0;
    margin-top: 0;
    text-align: center;
  }

  .subtext {
    color: #6f7477;
    flex: 0 0 auto;
    font-size: 1rem;
    line-height: 1.142;
    margin-bottom: 0;
    margin-top: 0;
    padding-top: 0.2857rem;
    text-align: center;
  }

  .subtext-content {
    flex: 0 0 auto;
    padding-top: 0.2857rem;
  }
}
