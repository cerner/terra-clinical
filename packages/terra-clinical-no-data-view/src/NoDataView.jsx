import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import IconNoData from './NoDataIcon';
import styles from './NoDataView.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The title of the no data view.
   */
  heading: PropTypes.string,
  /**
   * The subtext description of the no data view.
   */
  subtext: PropTypes.string,
  /**
   * The element to be dispaly in the subtext content container.
   */
  subtextContent: PropTypes.element,
  /**
   * Whether or not the no data glyph should be displayed.
   */
  isGlyphHidden: PropTypes.bool,
};

const defaultProps = {
  heading: '',
  subtext: '',
  subtextContent: undefined,
  isGlyphHidden: false,
};

const NoDataView = ({
    heading,
    subtext,
    subtextContent,
    isGlyphHidden,
    ...customProps
  }) => {
  let glyphSection;
  if (!isGlyphHidden) {
    const noDataIcon = <IconNoData />;
    glyphSection = <div className={cx('glyph')}>{noDataIcon}</div>;
  }

  let headingSection;
  if (heading) {
    headingSection = <p className={cx('heading')}>{heading}</p>;
  }

  let subtextSection;
  if (subtext) {
    subtextSection = <p className={cx('subtext')}>{subtext}</p>;
  }

  let subtextContentSection;
  if (subtextContent) {
    subtextContentSection = <div className={cx('subtext-content')}>{subtextContent}</div>;
  }

  return (
    <div {...customProps} className={cx('no-data-view', customProps.className)}>
      {glyphSection}
      {headingSection}
      {subtextSection}
      {subtextContentSection}
    </div>
  );
};

NoDataView.propTypes = propTypes;
NoDataView.defaultProps = defaultProps;

export default NoDataView;
