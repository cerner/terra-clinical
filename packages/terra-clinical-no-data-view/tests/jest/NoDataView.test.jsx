import React from 'react';
import NoDataView from '../../src/NoDataView';

const defaultVariety = <NoDataView />;

// Snapshot Tests
it('should render a default component', () => {
  const wrapper = render(defaultVariety);
  expect(wrapper).toMatchSnapshot();
});


// Prop Tests
it('should have the class no-data-view', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper.prop('className')).toContain('no-data-view');
});

it('should render a heading', () => {
  const noDataView = <NoDataView heading="test heading" />;
  const wrapper = render(noDataView);
  expect(wrapper).toMatchSnapshot();
});

it('should render a subtext', () => {
  const noDataView = <NoDataView subtext="test subtext." />;
  const wrapper = render(noDataView);
  expect(wrapper).toMatchSnapshot();
});

it('should render a with subtext content', () => {
  const noDataView = <NoDataView subtextContent={<p>test subtextContent</p>} />;
  const wrapper = render(noDataView);
  expect(wrapper).toMatchSnapshot();
});

it('should not render a glyph', () => {
  const noDataView = <NoDataView isGlyphHidden />;
  const wrapper = render(noDataView);
  expect(wrapper).toMatchSnapshot();
});
