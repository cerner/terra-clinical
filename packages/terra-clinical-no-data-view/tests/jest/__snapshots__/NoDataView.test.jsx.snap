// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render a glyph 1`] = `
<div
  class="no-data-view"
/>
`;

exports[`should render a default component 1`] = `
<div
  class="no-data-view"
>
  <div
    class="glyph"
  >
    <svg
      data-name="Layer 1"
      id="Layer_1"
      style="height:170px;width:170px;"
      viewbox="0 0 251.73 251.73"
    >
      <path
        d="M127.72,9.76a118,118,0,0,1,45.9,226.59A118,118,0,0,1,81.82,19a117.18,117.18,0,0,1,45.9-9.26m0-7.94A125.87,125.87,0,1,0,253.58,127.69,125.87,125.87,0,0,0,127.72,1.83Z"
        style="fill:#f4f4f4;"
        transform="translate(-1.85 -1.83)"
      />
      <path
        d="M94.29,116.54a5.13,5.13,0,0,1,4.56-2.77h83.78V103.2a5.25,5.25,0,0,0-5.2-5.2l-46.33.24a13.44,13.44,0,0,1-8.75-3.78l-9-9.22a13.44,13.44,0,0,0-8.75-3.78H74.37a5.25,5.25,0,0,0-5.2,5.2v79.42c0,.08,0,.15,0,.23a4.94,4.94,0,0,1,.55-2.08Z"
        style="fill:#e8e9ea;"
        transform="translate(-1.85 -1.83)"
      />
      <path
        d="M201.85,113.77h-103a5.13,5.13,0,0,0-4.56,2.77L69.73,164.22a4.94,4.94,0,0,0-.55,2.08,5,5,0,0,0,5,5.21H177.35a5.13,5.13,0,0,0,4.56-2.77l24.5-47.59A5.08,5.08,0,0,0,201.85,113.77Z"
        style="fill:#f4f4f4;"
        transform="translate(-1.85 -1.83)"
      />
    </svg>
  </div>
</div>
`;

exports[`should render a heading 1`] = `
<div
  class="no-data-view"
>
  <div
    class="glyph"
  >
    <svg
      data-name="Layer 1"
      id="Layer_1"
      style="height:170px;width:170px;"
      viewbox="0 0 251.73 251.73"
    >
      <path
        d="M127.72,9.76a118,118,0,0,1,45.9,226.59A118,118,0,0,1,81.82,19a117.18,117.18,0,0,1,45.9-9.26m0-7.94A125.87,125.87,0,1,0,253.58,127.69,125.87,125.87,0,0,0,127.72,1.83Z"
        style="fill:#f4f4f4;"
        transform="translate(-1.85 -1.83)"
      />
      <path
        d="M94.29,116.54a5.13,5.13,0,0,1,4.56-2.77h83.78V103.2a5.25,5.25,0,0,0-5.2-5.2l-46.33.24a13.44,13.44,0,0,1-8.75-3.78l-9-9.22a13.44,13.44,0,0,0-8.75-3.78H74.37a5.25,5.25,0,0,0-5.2,5.2v79.42c0,.08,0,.15,0,.23a4.94,4.94,0,0,1,.55-2.08Z"
        style="fill:#e8e9ea;"
        transform="translate(-1.85 -1.83)"
      />
      <path
        d="M201.85,113.77h-103a5.13,5.13,0,0,0-4.56,2.77L69.73,164.22a4.94,4.94,0,0,0-.55,2.08,5,5,0,0,0,5,5.21H177.35a5.13,5.13,0,0,0,4.56-2.77l24.5-47.59A5.08,5.08,0,0,0,201.85,113.77Z"
        style="fill:#f4f4f4;"
        transform="translate(-1.85 -1.83)"
      />
    </svg>
  </div>
  <p
    class="heading"
  >
    test heading
  </p>
</div>
`;

exports[`should render a subtext 1`] = `
<div
  class="no-data-view"
>
  <div
    class="glyph"
  >
    <svg
      data-name="Layer 1"
      id="Layer_1"
      style="height:170px;width:170px;"
      viewbox="0 0 251.73 251.73"
    >
      <path
        d="M127.72,9.76a118,118,0,0,1,45.9,226.59A118,118,0,0,1,81.82,19a117.18,117.18,0,0,1,45.9-9.26m0-7.94A125.87,125.87,0,1,0,253.58,127.69,125.87,125.87,0,0,0,127.72,1.83Z"
        style="fill:#f4f4f4;"
        transform="translate(-1.85 -1.83)"
      />
      <path
        d="M94.29,116.54a5.13,5.13,0,0,1,4.56-2.77h83.78V103.2a5.25,5.25,0,0,0-5.2-5.2l-46.33.24a13.44,13.44,0,0,1-8.75-3.78l-9-9.22a13.44,13.44,0,0,0-8.75-3.78H74.37a5.25,5.25,0,0,0-5.2,5.2v79.42c0,.08,0,.15,0,.23a4.94,4.94,0,0,1,.55-2.08Z"
        style="fill:#e8e9ea;"
        transform="translate(-1.85 -1.83)"
      />
      <path
        d="M201.85,113.77h-103a5.13,5.13,0,0,0-4.56,2.77L69.73,164.22a4.94,4.94,0,0,0-.55,2.08,5,5,0,0,0,5,5.21H177.35a5.13,5.13,0,0,0,4.56-2.77l24.5-47.59A5.08,5.08,0,0,0,201.85,113.77Z"
        style="fill:#f4f4f4;"
        transform="translate(-1.85 -1.83)"
      />
    </svg>
  </div>
  <p
    class="subtext"
  >
    test subtext.
  </p>
</div>
`;

exports[`should render a with subtext content 1`] = `
<div
  class="no-data-view"
>
  <div
    class="glyph"
  >
    <svg
      data-name="Layer 1"
      id="Layer_1"
      style="height:170px;width:170px;"
      viewbox="0 0 251.73 251.73"
    >
      <path
        d="M127.72,9.76a118,118,0,0,1,45.9,226.59A118,118,0,0,1,81.82,19a117.18,117.18,0,0,1,45.9-9.26m0-7.94A125.87,125.87,0,1,0,253.58,127.69,125.87,125.87,0,0,0,127.72,1.83Z"
        style="fill:#f4f4f4;"
        transform="translate(-1.85 -1.83)"
      />
      <path
        d="M94.29,116.54a5.13,5.13,0,0,1,4.56-2.77h83.78V103.2a5.25,5.25,0,0,0-5.2-5.2l-46.33.24a13.44,13.44,0,0,1-8.75-3.78l-9-9.22a13.44,13.44,0,0,0-8.75-3.78H74.37a5.25,5.25,0,0,0-5.2,5.2v79.42c0,.08,0,.15,0,.23a4.94,4.94,0,0,1,.55-2.08Z"
        style="fill:#e8e9ea;"
        transform="translate(-1.85 -1.83)"
      />
      <path
        d="M201.85,113.77h-103a5.13,5.13,0,0,0-4.56,2.77L69.73,164.22a4.94,4.94,0,0,0-.55,2.08,5,5,0,0,0,5,5.21H177.35a5.13,5.13,0,0,0,4.56-2.77l24.5-47.59A5.08,5.08,0,0,0,201.85,113.77Z"
        style="fill:#f4f4f4;"
        transform="translate(-1.85 -1.83)"
      />
    </svg>
  </div>
  <div
    class="subtext-content"
  >
    <p>
      test subtextContent
    </p>
  </div>
</div>
`;
